### PR TITLE
Fix #10626 adding possibilities for better updating store

### LINKED ISSUE #10626: extend possibilities to update state catalog and layers
## Description

For and issue of a client we need to extend possibilities to update state:
- update all layers at once
- update all catalog services at once

**What kind of improvement you want to add?** (check one with "x", remove the others)
- [ ] Minor changes to existing features
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### PR DESCRIPTION
**Pull request by MV88**
## Description

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

Fix #10626

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/actions/__tests__/catalog-test.js [hunk 1]
```diff
@@ -254,6 +256,14 @@ describe('Test correctness of the catalog actions', () => {
         expect(retval.type).toBe(ADD_CATALOG_SERVICE);
         expect(retval.service).toBe(service);
     });
+    it('updateCatalogServices', () => {
+        const services = [{}];
+        var retval = updateCatalogServices(services);
+
+        expect(retval).toExist();
+        expect(retval.type).toBe(UPDATE_CATALOG_SERVICES);
+        expect(retval.service).toBe(services);
```
Comment (MV88) [inline web/client/actions/__tests__/catalog-test.js:265]:
```suggestion
        expect(retval.service).toEqual(services);
```

File: web/client/actions/__tests__/layers-test.js [hunk 2]
```diff
@@ -240,6 +242,14 @@ describe('Test correctness of the layers actions', () => {
         expect(action.options).toEqual({opacity: 0.5});
     });
 
+    it('updateAllLayers', () => {
+        const layers = [{}];
+        const action = updateAllLayers(layers);
+        expect(action).toExist();
+        expect(action.type).toBe(UPDATE_ALL_LAYERS);
+        expect(action.layers).toBe(layers);
```
Comment (MV88) [inline web/client/actions/__tests__/layers-test.js:250]:
```suggestion
        expect(action.layers).toEqual(layers);
```

File: web/client/actions/layers.js [hunk 3]
```diff
@@ -36,7 +36,19 @@ export const FILTER_LAYERS = 'LAYERS:FILTER_LAYERS';
...
+export const UPDATE_ALL_LAYERS = 'LAYERS:UPDATE_ALL_LAYERS';
 
+/**
+ * full replacement of layers in layers state
+ * @param {object[]} layers the new layers to replace in the state
+ * @returns
+ */
+export function updateAllLayers(layers) {
+    return {
+        type: UPDATE_ALL_LAYERS,
```
Comment (allyoucanmap) [inline web/client/actions/layers.js:48]: Based also on the jsdoc this operation represent a replacement so I would call this REPLACE_LAYERS/replaceLayers

**Review by allyoucanmap — APPROVED** [review]

**Comment by allyoucanmap** [comment]
@MV88 @ElenaGallo this improvements are related to new actions for development so I don't think there are test to be done

**Comment by MV88** [comment]
> @MV88 @ElenaGallo this improvements are related to new actions for development so I don't think there are test to be done

correct

**Pull request merged**